src/gt4py/next/program_processors/runners/gtfn_cpu.py

```diff
@@ -67,44 +98,50 @@ class GTFNExecutor(ppi.ProgramExecutor):
     enable_itir_transforms: bool = True  # TODO replace by more general mechanism, see https://github.com/GridTools/gt4py/issues/1135
     use_imperative_backend: bool = False
 
-    def __call__(self, program: itir.FencilDefinition, *args: Any, **kwargs: Any) -> None:
-        """
-        Execute the iterator IR program with the provided arguments.
-
-        The program is compiled to machine code with C++ as an intermediate step,
-        so the first execution is expected to have a significant overhead, while subsequent
-        calls are very fast. Only scalar and buffer arguments are supported currently.
-
-        See ``ProgramExecutorFunction`` for details.
-        """
-
-        def convert_args(inp: Callable) -> Callable:
-            def decorated_program(*args):
-                return inp(
-                    *[convert_arg(arg) for arg in args],
-                    *extract_connectivity_args(kwargs["offset_provider"]),
-                )
+    # TODO(tehrengruber): The otf workflow currently takes the entire closure, i.e. the
+    #  fencil and its arguments, instead of the parts it actually needs to compile the program.
+    #  The hash function for the caching currently only uses a subset of the closure and implicitly
+    #  relies on the workflow to only use that information. As this is dangerous the caching is
+    #  disabled by default and should be considered experimental.
```

Comment:
Again, what is the todo item here? Is it to restrict the information the workflow gets? Is it to improve the hashing? Is it to determine which is easier and then implement that? Or is it to redesign on a higher level?